### DISCUSSION TITLE
Player open/close book event

### DISCUSSION THREAD

**Discussion by KioProject123** [PaperMC/Paper#11181] [Ideas]
### Is your feature request related to a problem?

Cancel the player opening a book, or perform an action when the player opens a book.

### Describe the solution you'd like.

Add an event when the player to open/close a book.

### Describe alternatives you've considered.

Listen for PlayerInteractEvent and check that the player is holding a book and that the player is not interacting with other blocks (such as opening a chest, a door, etc.)
It is impossible to check the behavior of all block interactions.

**Comment by electronicboy** [comment]
I'm not sure that the server knows if a player has a book open or not? [too dead too check]

**Reply by KioProject123** [reply]
Yes, the server sends the player a `ClientboundOpenBookPacket`

**Reply by electronicboy** [reply]
That only works for written books, not sure we hear about closing, however; unless this changed recently

**Comment by Malfrador** [comment]
It does. Right-clicking a book causes the server to send a packet that then opens the GUI on the client.

That also means that the PlayerInteractEvent would be the correct event for this. You can easily check for only Item interactions using [useInteractedBlock](https://jd.papermc.io/paper/1.21/org/bukkit/event/player/PlayerInteractEvent.html#useInteractedBlock())() and [useItemInHand](https://jd.papermc.io/paper/1.21/org/bukkit/event/player/PlayerInteractEvent.html#useItemInHand())(), a new event shouldn't be necessary for this.

**Reply by KioProject123** [reply]
I don't understand how to distinguish whether the player is opening a book or interacting with a block. 
When the player holds a book and right click with a grass block or a chest (the former opens the book, the latter opens the chest), useInteractedBlock() always returns ALLOW and useItemInHand() always returns DEFAULT

**Reply by NonSwag** [reply]
Sneaking invokes a hand interaction even if the block has an interaction
not sneaking and right-clicking a block that has an interaction triggers the block interaction
see #11207
